File: question1.js
function solution(A) {
    A.sort((a, b) => a - b);

    let longestSpike = 1;
    let currentSpike = 1;

    for (let i = 1; i < A.length; i++) {
        if (A[i] > A[i - 1]) {
            currentSpike++;
            longestSpike = Math.max(longestSpike, currentSpike);
        } else {
            currentSpike = 1;
        }
    }

    return longestSpike;
}

File: question1.rb
def solution(a)
    a.sort!
  
    longest_spike = 1
    current_spike = 1
  
    (1...a.length).each do |i|
      if a[i] > a[i - 1]
        current_spike += 1
        longest_spike = [longest_spike, current_spike].max
      else
        current_spike = 1
      end
    end
  
    longest_spike
  end